### PR TITLE
Acceptors changes for learner

### PR DESCRIPTION
**Pull request by dvr0**
(basically copied Konstantin code into acceptor lol)

- resending 2B message when receiving phase 4
- Acceptor timeout: start after sending 1B, if phase 2A message not received ask to restart consensus

⚠️ Things to double check:
- line 156
- is the phase 3 condition (accept if quorum of processes has c-rnd = v-rnd)? Do we care?
- What if init value is sent for learner timeout on acceptor?
- Quorum from same process issue - is there a case when we can get a quorum from the same acceptor combining timeout and message delay? Again, do we care?
- what happens when validity is broken in the learner?

### CODE REVIEW

File: our-paxos/paxos.py [hunk 1]
```diff
@@ -106,44 +103,69 @@ def acceptor(config, id):
...
     r = mcast_receiver(config['acceptors'])
     s = mcast_sender()
+
+    # timeout when 2A messages are lost
+    def acceptor_timeout(id):
+        sleep(TIMEOUT)
+        state = paxos_instances[id]
+
+        # have we received a phase 2A message?
+        while (state['v-rnd'] != 0):
```
Comment (BritikovKI) [inline our-paxos/paxos.py:113]: You are going just to send this request infinitely until you receive some message back, which might be a little bit of an overkill

File: our-paxos/paxos.py [hunk 2]
```diff
@@ -106,44 +103,69 @@ def acceptor(config, id):
...
-                print("i: {} p: {}, state: {}".format(instance, phase, state))
+                print("i: {} p: {}, state: {}".format(id, phase, state))
 
                 # encode and send phase 2B to proposer
-                msg = paxos_encode([instance, phase, state['v-rnd'], state['v-val']])
+                msg = paxos_encode([id, phase, state['v-rnd'], state['v-val']])
                 s.sendto(msg, config['learners'])
+        
+        elif phase == 4: # received "resend 2B" request from learner timeout
+            msg = paxos_encode([id, 2, state['v-rnd'], state['v-val']])
```
Comment (BritikovKI) [inline our-paxos/paxos.py]: Where do you get state['v-rnd']? What if you have few instances going simultanuosly?

File: our-paxos/paxos.py (hunk 1 above)
Comment (dvr0) [reply our-paxos/paxos.py:113]: Yes. How else though? 😄

File: our-paxos/paxos.py (hunk 2 above)
Comment (dvr0) [reply our-paxos/paxos.py]: discussed this

**Pull request merged**